src/Build/Evaluation/ItemSpec.cs

```diff
@@ -498,5 +498,7 @@ public GlobFragment(string textFragment, string projectDirectory)
             : base(textFragment, projectDirectory)
         {
         }
+
+        public bool IsFullFileSystemScan => (TextFragment.StartsWith(@"\**\") || TextFragment.StartsWith(@"/**/"));
```

Comment:
Think degenerate cases like `/**\` and `\**/` are worth catching?

Reply:
Those couldn't lead to a full file system scan, could they? For `/**\` for example, a Windows OS wouldn't know what to do with the first /, and a mac wouldn't know what to do with the second, so neither would interpret the whole correctly.

Reply:
Fixed by detecting all 4 variations, thanks!

> For /**\ for example, a Windows OS wouldn't know what to do with the first /, and a mac wouldn't know what to do with the second, so neither would interpret the whole correctly.

I believe that slashes are normalized to match the platform MSBuild is running on so you can, in fact, arbitrarily mix & match forward and backward slashes.